pkg/engine/handlers/validation/validate_pss.go

```diff
@@ -119,15 +131,100 @@ func (h validatePssHandler) Process(
 		if len(pssChecks) == 0 && err == nil {
 			podSecurityChecks.Checks = pssChecks
 			logger.V(3).Info("policy rule is skipped due to policy exceptions", "exceptions", keys)
-			return resource, handlers.WithResponses(
-				engineapi.RuleSkip(rule.Name, engineapi.Validation, "rule is skipped due to policy exceptions "+strings.Join(keys, ", "), rule.ReportProperties).WithExceptions(matchedExceptions).WithPodSecurityChecks(podSecurityChecks),
-			)
+			return resource, engineapi.RuleSkip(rule.Name, engineapi.Validation, "rule is skipped due to policy exceptions "+strings.Join(keys, ", "), rule.ReportProperties).WithExceptions(matchedExceptions).WithPodSecurityChecks(podSecurityChecks)
 		}
 		msg := fmt.Sprintf(`Validation rule '%s' failed. It violates PodSecurity "%s:%s": %s`, rule.Name, podSecurity.Level, podSecurity.Version, pss.FormatChecksPrint(pssChecks))
-		return resource, handlers.WithResponses(
-			engineapi.RuleFail(rule.Name, engineapi.Validation, msg, rule.ReportProperties).WithPodSecurityChecks(podSecurityChecks),
+		ruleResponse := engineapi.RuleFail(rule.Name, engineapi.Validation, msg, rule.ReportProperties).WithPodSecurityChecks(podSecurityChecks)
+		allowExisitingViolations := rule.HasValidateAllowExistingViolations()
+		if engineutils.IsUpdateRequest(policyContext) && allowExisitingViolations {
+			logger.V(4).Info("is update request")
+			priorResp, err := h.validateOldObject(ctx, logger, policyContext, resource, rule, engineLoader, exceptions)
+			if err != nil {
+				return resource, engineapi.RuleError(rule.Name, engineapi.Validation, "failed to validate old object", err, rule.ReportProperties)
```

Comment:
Do we want to error it out if `allowExisitingViolations=true` for an UPDATE request? If I allowed that as a user, I wouldn't expect any failure blocks my update request.

This applies to all types of validate rules.

Reply:
agreed, fixing